### ISSUE TITLE
Directory reorganization needed

### ISSUE THREAD

**Issue by HazenBabcock** [ZhuangLab/storm-analysis#3]
To reduce name space pollution most things should be moved to a sub-directory of a new folder called "storm_analysis", so the new layout would be something like:

storm_analysis/3d_daostorm
storm_analysis/decon_storm

..

**Comment by HazenBabcock** [comment]
We also need a better name for the 3d_daostorm directory to make it easier to import.

**Comment by Dschoni** [comment]
Wouldn't it be appropriate, to reorganize the whole project in a python module-like structure with according __init__.py files as well?

**Comment by HazenBabcock** [comment]
I renamed everything basically as described above. And I moved the 3D-DAOSTORM project to the daostorm_3d directory.

Can you describe a bit more what you mean regarding init.py files?

**Comment by Dschoni** [comment]
init.py (or more exact __init__.py files) are used to
mark a directory as a module. At some point, I feel that
end users want to import packages rather than setting
folders as working directories. Furthermore, it would be
a step towards distributing the whole project as a package/module
together with setup.py files that handle the installation/compilation
of external c-modules on the fly.

Am 2016-09-10 03:32, schrieb Hazen Babcock:

**Comment by HazenBabcock** [comment]
I believe that all the relevant folders already have __init__.py files in them, so I guess I still don't understand your suggestion. You only need to add the root storm-analysis directory to your Python path and after that you can access pretty much anything in this project using import statements.

I'm unsure about the best way to distribute this project because only parts of it are library functions. Also, my experience is that setup.py always fails when you are using the MinGW compiler on Windows. I've never figured out how to pass the correct arguments to setup.py to get it to work.

**Comment by Dschoni** [comment]
The relevant init.py files are empty.
In order to distribute a package, all relevant imports
used for a module are defined in these init files, so that
f.e. submodules are imported automatically. Furthermore, there
should be an setup.py file in the root directory, that takes care
of compilation on the client and installing the package.
This is standard deployment for python packages rather than distributing
.sh scripts and the need to add something to path.

Making stuff windows compatible is another big thing. There are some
issues compiling against different versions of the python runtime and 
basically
you have three options.

-Using the GNU toolchain with mingw (which I also don't really like)
-Compiling with Visual Studio (Which forces consistent c95 programming)
-Distributing precompiled dynamic libraries (which is kind of a 
compromise...)

I see the main problem being the Fortran calls to LAPACK, which makes 
stuff a
lot harder to cross-compile against. But as scipy also links to those 
libraries,
someone, somewhere should have solved that problem already.

Am 2016-09-24 03:43, schrieb Hazen Babcock:

**Comment by hadim** [comment]
Now that `storm_analysis` is a Python module you could add a `setup.py` file so it makes it easier for people to install. However they will still need to manually compile C extension which is a pain on Windows.

So in addition to the `setup.py` file you could also build a conda-forge recipe (https://conda-forge.github.io/) so it makes it possible to distribute binary packages of `storm_analysis` for every platform (Windows, Linux, OSX). No more compilation needed, just install it with `conda install storm_analysis` and you're done.

I can help setting up the build system if needed.

**Comment by hadim** [comment]
A PR to add `setup.py` is available here https://github.com/ZhuangLab/storm-analysis/pull/12

**Comment by HazenBabcock** [comment]
This is now in master.
